### PR TITLE
chore: Outlook `processBusyTimes` iterate busy times once

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR improves the performance of how we process the busy times from Outlook by only iterating through the times once instead of twice.

Fixes # (issue)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [x] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- Connect an Outlook calendar with busy times
- Select those calendars to check availability on
- When booking an event ensure that those events are blocked

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/app-store/office365calendar/lib/CalendarService.ts [hunk 1]
```diff
@@ -454,12 +454,13 @@ export default class Office365CalendarService implements Calendar {
       (acc: BufferedBusyTime[], subResponse: { body: { value?: BodyValue[]; error?: Error[] } }) => {
         if (!subResponse.body?.value) return acc;
         return acc.concat(
-          subResponse.body.value
-            .filter((evt) => evt.showAs !== "free" && evt.showAs !== "workingElsewhere")
-            .map((evt) => ({
```
Comment (joeauyeung) [inline packages/app-store/office365calendar/lib/CalendarService.ts:459]: If we chain `.filter` and `.map` we are essentially iterating through the array twice. By replacing this with a `.reduce` we're able to filter through values while only iterating through the array once.

**Review by alannnc — APPROVED** [review]

File: packages/app-store/office365calendar/lib/CalendarService.ts (hunk 1 above)
Comment (hariombalhara) [reply packages/app-store/office365calendar/lib/CalendarService.ts:459]: `filter` and `map` usage keeps the code clean and simple to understand. Unless the array is huge it doesn't really matter performance wise how many times we iterate over it.

What's the performance improvement in here?

**Review by hariombalhara — APPROVED** [review]
LGTM! Left one comment to get clarity

File: packages/app-store/office365calendar/lib/CalendarService.ts (hunk 1 above)
Comment (PeerRich) [reply packages/app-store/office365calendar/lib/CalendarService.ts:459]: lets do this in a follow up PR

Comment (joeauyeung) [reply packages/app-store/office365calendar/lib/CalendarService.ts:459]: @hariombalhara there was a user that was experiencing a long load time for availability from their Outlook calendar. From what I gathered the payload was very large. I think most of the time the busy times array is small enough that it's fine to do the `filter` and `map` but in this scenario, we're running into the theoretical O(2n) vs O(n).

Comment (hariombalhara) [reply packages/app-store/office365calendar/lib/CalendarService.ts:459]: Thanks. Makes sense then.

**Pull request merged**